### PR TITLE
Setup backend project

### PR DESCRIPTION
**Pull request by milensski**
### Description
This pull request aims to set up the project backend using Django/Rest Framework and PostgreSQL. 
It includes the initial project structure and database configuration. 

### Changes Made
- Initialized a Django project. 
- Installed and configured Django Rest Framework in settings.py
- Added requirements.txt
- Added docker-compose file for the DB setup 
  > the compose file includes setting for pgadmin4 if needed.
- Added .gitignore folders to the top-level file

**Please Note** : There are no initial API endpoints configured. 

#### Comment :  
We have defined the project folder structure to be divided into smaller applications which ensures scalability later on.
- apps/accounts - will take care of user accounts
- apps/card - will be the application serving the endpoints for the cards
- apps/user_profiles - not sure for now if this is necessary for extending the user profile in accounts

> **To test the backend it needs .env files for the settings.py variables and for the docker-compose file**

### CODE REVIEW

**Comment by julkascript** [comment]
Good start, guys. I suggest simplifying the structure because when the code gets bigger, the structure can put a toll on us and our work. 

Currently what we have is:
```
.
└── backend
  └── cardflow    # <- This directory holds all files just like it's parent, so I don't see a point in keeping it.
      ├── apps   
        └── app1
        └── app2
      ├── cardflow
      ├── docker-compose.yml
      ├── manage.py
      └── requirements.txt
```
I suggest simplifying the structure to:
```
.
└── backend  
      ├── app1
      ├── app2
      ├── cardflow
      ├── docker-compose.yml
      ├── manage.py
      └── requirements.txt
```

**Comment by julkascript** [comment]
- [ ] Fix the conflict in the `.gitignore` file by accepting both changes

File: backend/cardflow/requirements.txt
Comment (julkascript) [inline backend/cardflow/requirements.txt:1]: - [ ] Freeze all requirements in the requirements file

Comment (milensski) [reply backend/cardflow/requirements.txt:1]: fixed

**Comment by milensski** [comment]
Hi,
We have discussed the structure and we agree on the suggested by you.
The project is now formatted as the below scheme:
.
└── backend  
      ├── accounts
      ├── yugi
      ├── cardflow
      ├── docker-compose.yml
      ├── manage.py
      └── requirements.txt

The card app is renamed to yugi. The idea is to have different apps for the different games.

**Comment by julkascript** [comment]
- [ ] Freeze the dependencies in requirements.txt

**Comment by milensski** [comment]
> * [X]  Freeze the dependencies in requirements.txt

> [!NOTE]
> The dependencies are freezed into requirements.txt

I have changed the name of the app **yugi** to `card`

**Review by julkascript — APPROVED** [review]

**Pull request merged**